tests/fileVersionTest.js

```diff
@@ -201,32 +179,62 @@ describe('Main: Currently testing file versions management,', function () {
-      await PropfindFileVersionOfExistentFiles(provider)
-      await getFileVersionContents(provider)
-      await restoreFileVersion(provider)
+      await getCapabilitiesInteraction(provider, config.testUser, config.testUserPassword)
+      await getCurrentUserInformationInteraction(provider, config.testUser, config.testUserPassword)
+      provider
+        .given('the user is recreated', {
+          username: config.testUser,
+          password: config.testUserPassword
+        })
+        .given('file exists', {
+          fileName: versionedFile,
+          username: config.testUser,
+          password: config.testUserPassword
+        })
+        // re-upload the same file to create a new version
+        .given('file exists', {
+          fileName: versionedFile,
+          username: config.testUser,
+          password: config.testUserPassword
+        })
+        .given('file version link is returned', {
+          fileName: versionedFile,
+          username: config.testUser,
+          password: config.testUserPassword,
+          number: 1
+        })
+        .given('provider base url is returned')
+      provider
+        .uponReceiving('Restore file versions')
+        .withRequest({
+          method: 'COPY',
+          path: MatchersV3.fromProviderState(
+            // eslint-disable-next-line no-useless-escape,no-template-curly-in-string
+            '\${versionLink}',
+            `/remote.php/dav/meta/${fileInfo.id}/v/${fileInfo.versions[0].versionId}`
+          ),
+          headers: {
+            authorization: getAuthHeaders(config.testUser, config.testUserPassword),
+            Destination: MatchersV3.fromProviderState(
+              `\${providerBaseURL}${destinationWebDavPath}`,
+              `${config.owncloudURL}${destinationWebDavPath}`
+            )
+          }
+        })
+        .willRespondWith({
+          status: 204
+        })
 
       return provider.executeTest(async () => {
-        const oc = createOwncloud()
+        const oc = createOwncloud(config.testUser, config.testUserPassword)
         await oc.login()
-        return oc.fileVersions.listVersions(fileInfo.id).then(versions => {
-          expect(versions.length).toEqual(2)
-          expect(versions[0].getSize()).toEqual(2)
-          expect(versions[1].getSize()).toEqual(1)
-          oc.fileVersions.restoreFileVersion(fileInfo.id, fileInfo.versions[0].versionId, versionedFile).then(status => {
-            expect(status).toBe(true)
-            oc.files.getFileContents(versionedFile).then(content => {
-              expect(content).toBe(fileInfo.versions[0].content)
-              done()
-            })
-          }).catch(reason => {
-            fail(reason)
-            done()
-          })
+        return oc.fileVersions.restoreFileVersion(fileInfo.id, fileInfo.versions[0].versionId, versionedFile).then(status => {
+          expect(status).toBe(true)
+        }).catch(reason => {
+          fail(reason)
```

Comment:
why did you strip down this test?

Reply:
this test is about restoring file version, no need to check the content here as we are testing against a mock server anyway see 86ae9a09984badee76c87525eb865c630135c900